tests/auth/registration-modal.spec.ts

```diff
@@ -0,0 +1,428 @@
+import { Locator, Page, expect, test } from "@playwright/test";
+
+// ==========================================================================
+test.describe("Sign up modal", () => {
+  // -------------------------------------------------------------------------
+  test("Sign up modal visible", async ({ page }) => {
+    await page.goto("/");
+
+    const signUpButton = page.locator("button", {
+      hasText: "Sign up",
+    });
+
+    await expect(
+      signUpButton,
+      "Sign up button should be visible"
+    ).toBeVisible();
+    await expect(
+      signUpButton,
+      "Sign up button should be enabled"
+    ).toBeEnabled();
+
+    signUpButton.click();
+
+    const signUpModal = page.locator("app-signup-modal");
+
+    await expect(signUpModal, "Sign up modal should be visible").toBeVisible();
+  });
+
+  // ==========================================================================
+  test.describe("Registration form", () => {
+    // -------------------------------------------------------------------------
+    const redColor = "rgb(220, 53, 69)";
+    const errorMessagePath = "div.invalid-feedback > p";
+    const nameInputPath = "input#signupName";
+    const lastNameInputPath = "input#signupLastName";
+    const emailInputPath = "input#signupEmail";
+    const passwordInputPath = "input#signupPassword";
+    const repeatPasswordInputPath = "input#signupRepeatPassword";
+    // -------------------------------------------------------------------------
+    const changingBorderColor = () =>
```

Comment:
in fact, it is a delay. Bad practice. It can be used but if other normal ways did not help. In this case, it is inappropriate

Reply:
use web first assertions instead